### PR TITLE
add treasuries pallet

### PR DESCRIPTION
**Pull request by brenzi**
* adds treasuries pallet and bumps democracy to add `SpendNative` proposals
* adds cli tooling for the above
* adds rpc

## testing

```
./target/release/encointer-node-notee --dev --enable-offchain-indexing true

./bootstrap_demo_community.py --signer //Bob
nctr-dev next-phase
nctr-dev next-phase
nctr-dev next-phase
### fill treasury 
nctr-dev get-treasury --cid sqm1v79dF6b
# 5CWoc3mGF9VEnuZzBbPWxhKPvY743AGwxUbvkYQHS8yWZbem
nctr-dev transfer //Alice 5CWoc3mGF9VEnuZzBbPWxhKPvY743AGwxUbvkYQHS8yWZbem 1000000000000
nctr-dev balance 5CWoc3mGF9VEnuZzBbPWxhKPvY743AGwxUbvkYQHS8yWZbem
# 1000000000000
nctr-dev submit-spend-native-proposal //Alice //Bob 123456789012 --cid sqm1v79dF6b
nctr-dev list-proposals
# verify electorate is 3
nctr-dev vote //Alice 1 aye sqm1v79dF6b_1
# await confirming 
nctr-dev update-proposal-state //Alice 1
nctr-dev list-proposals
# verify approval
nctr-dev next-phase
# verify enactment
# should see
# encointerDemocracy.ProposalEnacted
# encointerTreasuries.SpentNative
# balances.transfer to Bob
nctr-dev balance 5CWoc3mGF9VEnuZzBbPWxhKPvY743AGwxUbvkYQHS8yWZbem
# treasury balance went down
```

### CODE REVIEW

File: client/src/commands/encointer_treasuries.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	rt.block_on(async {
+		let api = get_chain_api(matches).await;
+
+		let maybecid = if let Some(cid) = matches.cid_arg() {
+			Some(api.verify_cid(cid, None).await)
+		} else {
+			None
+		};
+		let treasury = api.get_community_treasury_account_unchecked(maybecid).await.unwrap();
+		// only print plain businesses to be able to parse them in python scripts
```
Comment (clangenb) [inline client/src/commands/encointer_treasuries.rs]: I guess the comment is a copy paste error.

File: runtime/src/weights/mod.rs [hunk 2]
```diff
@@ -9,4 +9,4 @@ pub mod pallet_encointer_democracy;
 pub mod pallet_encointer_faucet;
 pub mod pallet_encointer_reputation_commitments;
 pub mod pallet_encointer_scheduler;
-pub mod pallet_treasury;
+// pub mod pallet_treasury;
```
Comment (clangenb) [inline runtime/src/weights/mod.rs:12]: Intended?

**Review by clangenb — COMMENTED** [review]
Two tiny remarks, looks good otherwise

File: runtime/src/weights/mod.rs (hunk 2 above)
Comment (brenzi) [reply runtime/src/weights/mod.rs:12]: yes. it causes warnings and we're not actually using it right now. will reconsider when we benchamrk the next time

**Pull request merged**